compiler/annotation/mod.rs

```diff
@@ -203,6 +203,16 @@ typedb_error!(
             variable: Variable,
             source_span: Option<Span>,
         ),
+        DetectedUnsatisfiableEdge(
```

Comment:
Quick questions
- does this work with links with player/relation/role player 3 way component edge? How does it look?
- I'm not a huge fan of the user seeing '$0 has $1` like you did in the PR description. I would probably just delete this part and rely on the pointer into the query that indicates where it failed to be honest! Less redundancy.

Reply:
They're all binary edges in TypeInferenceGraph.
```
INF11 
[INF11] Type-inference was unable to find compatible types for the pair of variables 'p' & '_anonymous' across a constraint. Types were:
- p: [person]
- _anonymous: [friendship:friend]
Caused: [QUA1] Type inference error while compiling query annotations.
Caused: [QEX8] Error analysing query.
Near 1:48
-----
--> match $p isa person; $f isa friendship, links (friend: $p);
                                                   ^
-----
```

Reply:
Actually, I think I can fix that `_anonymous` to be a label if Links can be refactored to take a Label as role_type vertex

Reply:
ok _anonymous is better than $0, thanks